app/util/getICOTokens.js

```diff
@@ -1,22 +1,24 @@
-export const getICOTokens = () =>
-  // Import List. For now return hardcoded
-  [
-    {
-      token: 'RPX',
-      scriptHash: 'AEdAeaa7421adkdaslkea2321qd',
-      supply: 11333311145,
-      name: 'Red Pulse'
-    },
-    {
-      token: 'ONT',
-      scriptHash: 'AEdAeaa7421adkdaslkea2321qd',
-      supply: 11333311145,
-      name: 'Ontology'
-    },
-    {
-      token: 'RNG',
-      scriptHash: 'AEdAeaa7421adkdaslkea2321qd',
-      supply: 11333311145,
-      name: 'Random Test Token'
-    }
-  ]
+import { get } from 'lodash'
+import store from '../store/configureStore'
+
+import ICOTokenList from '../../ICOTokens.json'
+
+export const getICOTokens = () => {
+  const state = store.getState()
+  /* eslint-disable-next-line */
+  const tokens = get(state, 'spunky.settings.data.tokens')
+  if (!Array.isArray(tokens)) return null
+  const userGeneratedTokens = tokens.filter(token => token.isUserGenerated)
+
+  const combinedTokenList = [
+    ...userGeneratedTokens,
+    ...ICOTokenList.ICOTokens
+  ].map(token => ({
+    token: token.symbol,
```

Comment:
I dont think user added scripts will ever have a symbol at this point we probably want to call `getToken`  here http://cityofzion.io/neon-js/docs/en/api/plugin/nep5.html#methods to obtain symbol

Reply:
we already do this in balancesActions.js for example...